### PR TITLE
{numlib}[intel/2021a] Hypre v2.21.0, MUMPS v5.4.0, PETSc v3.15.1, ... w/ Python 3.9.5

### PR DESCRIPTION
**Pull request by stevenvdb**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/p/PETSc/PETSc_remove-mpiicc-show-quotes.patch [hunk 1]
```diff
@@ -0,0 +1,12 @@
+diff -Nru config/PETSc/Configure.py.back config/PETSc/Configure.py`
```
Comment (Micket) [inline easybuild/easyconfigs/p/PETSc/PETSc_remove-mpiicc-show-quotes.patch:1]: Could you please add a line of description of what this patch fixes (and who the author is (you presumably))

**Review by Micket — DISMISSED** [review]
lgtm

**Comment by Micket** [comment]
@boegelbot please test @ generoso

**Comment by Micket** [comment]
Test report by @Micket
**FAILED**
Build succeeded for 4 out of 6 (4 easyconfigs in total)
vera-c2 - Linux Rocky Linux 8.6, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/435d8df2cd00f995c5111e963345aafa for a full test report.

**Comment by Micket** [comment]
```
icc: error #10106: Fatal error in /apps/Test2/software/intel-compilers/2021.2.0/compiler/2021.2.0/linux/bin/intel64/../../bin/intel64/mcpcom, terminated by kill signal
compilation aborted for /dev/shm/SuiteSparse/5.10.1/intel-2021a-METIS-5.1.0/SuiteSparse-5.10.1/GraphBLAS/Source/GB_AxB_saxpy_generic.c (code 1)
```
hm, not good, and no obvious workaround or solution to me. generoso build seems to have had probably the same issue

**Comment by stevenvdb** [comment]
> ```
> icc: error #10106: Fatal error in /apps/Test2/software/intel-compilers/2021.2.0/compiler/2021.2.0/linux/bin/intel64/../../bin/intel64/mcpcom, terminated by kill signal
> compilation aborted for /dev/shm/SuiteSparse/5.10.1/intel-2021a-METIS-5.1.0/SuiteSparse-5.10.1/GraphBLAS/Source/GB_AxB_saxpy_generic.c (code 1)
> ```
> 
> hm, not good, and no obvious workaround or solution to me. generoso build seems to have had probably the same issue

I checked my local build, and SuiteSparse used about 120GB at some point during the installation. It seems reasonable to see "terminated by kill signal" if the test job runs out of memory. Is there a way to check this and/or rerun the test jobs with more available memory?

**Comment by Micket** [comment]
That's hefty. I lost track of my old build log, but i think it was during hte compilation phase, which we should consider limiting the number of parallel builds to conserve memory. I see older easyconfigs had `parallel = 1` (perhaps for this reason? there was no comments telling why unfortunately).
We could go a bit higher then just 1 to speed things up, maybe
```
parallel = 8
```

**Comment by stevenvdb** [comment]
> That's hefty. I lost track of my old build log, but i think it was during hte compilation phase, which we should consider limiting the number of parallel builds to conserve memory. I see older easyconfigs had `parallel = 1` (perhaps for this reason? there was no comments telling why unfortunately). We could go a bit higher then just 1 to speed things up, maybe
> 
> ```
> parallel = 8
> ```

I did a serial build, but it seems there is one `mcpcom` process that is responsible for this high memory usage. I attached the [file](https://gist.github.com/stevenvdb/e56841e7d615873aff2d6caf728c53d9) that is being compiled while the memory usage is high, I don't really know what is happening...

**Comment by boegel** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

**Comment by boegel** [comment]
@boegelbot please test @ generoso
CORE_CNT=16
EB_ARGS="--parallel 4"

**Comment by stevenvdb** [comment]
The very high memory consumption by icc is fixed by turning off the generation of optimization reports in 5af7bc9f581ce7030480c7238b7e6a734b8de6dc

One thing I noticed is that GraphBLAS will not respect the number of parallel processes EasyBuild tells `make` to use. Instead, it uses the variable `$JOBS` and if that is not set it default to 8 (see https://github.com/DrTimothyAldenDavis/SuiteSparse/blob/v5.10.1/GraphBLAS/Makefile). If there is a better way to deal with this than what I did https://github.com/easybuilders/easybuild-easyconfigs/blob/5f2dd890807b9b70758c70260998a2a2ae4c5912/easybuild/easyconfigs/s/SuiteSparse/SuiteSparse-5.10.1-intel-2021a-METIS-5.1.0.eb#L34-L37, let me know.

**Comment by branfosj** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 12 out of 12 (4 easyconfigs in total)
bear-pg0105u03a.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/d41d8bc2fe82456fb2005e26a2411d14 for a full test report.

**Review by branfosj — APPROVED** [review]

**Comment by branfosj** [comment]
Going in, thanks @stevenvdb!

**Pull request merged**